### ISSUE TITLE
nixos/grub: can only be installed into an EFI partition mounted /boot 

### ISSUE THREAD

**Issue by Atemu** [NixOS/nixpkgs#202036]
### Describe the bug

Having the boot/EFI partition anywhere but /boot results in the GRUB installer ignoring it and then putting bad paths in grub.cfg because it thinks /boot is part of the rootfs:

```
  linux ($drive1)/Root/boot/kernels/f6825zhvwwhwfchvf7br0pp5n96wpk41-linux-5.15.77-Image init=/nix/store/cn9k3fjs5vrdb38xd9h5pa2glifmx9ba-nixos-system-URANOS-22.11.git.c6818ce7b1f/init transparent_hugepage=always loglevel=4
  initrd ($drive1)/Root/boot/kernels/fcp98nnb5hij9fya4bvxnjzmzly2258g-initrd-linux-5.15.77-initrd
```

(`Root` is my rootfs btrfs subvolume, this would be correct if I didn't have a separate boot subvolume.)

Correct paths would be:

```
  linux ($drive1)//kernels/f6825zhvwwhwfchvf7br0pp5n96wpk41-linux-5.15.77-Image init=/nix/store/3kgj243l5s0n7qkb1pl2628bm3fn6kp6-nixos-system-URANOS-22.11.git.c6818ce7b1f/init transparent_hugepage=always loglevel=4
  initrd ($drive1)//kernels/fcp98nnb5hij9fya4bvxnjzmzly2258g-initrd-linux-5.15.77-initrd
```

This even happens when `/boot` is a symlink to a different place.

### Steps To Reproduce
1. Have nothing mounted at `/boot`
2. Have `boot.loader.efi.efiSysMountPoint` set to something other than `/boot`
3. Have a fat32 partition mounted at that other place

### Expected behavior

It should install bootloader and kernels to the declared mount point rather than trying to use `/boot`.

### Notify maintainers

cc @nh2 

cc @samueldr (You have a setup similar to mine and probably also have your boot directory mounted elsewhere)

### Metadata

667e5581d16745bcda791300ae7e2d73f49fff25

**Comment by samueldr** [comment]
> cc @samueldr (You have a setup similar to mine and probably also have your boot directory mounted elsewhere)

Nah, `/boot` is hidden with gobohide :). I have a plain setup with the ESP mounted at /boot.

But I agree this is a deficiency and most likely requires fixing.
